### ISSUE TITLE
Todo: Add a save & load feature

### ISSUE THREAD

**Issue by mindstorm8191** [mindstorm8191/danidle#9]
This game is growing, and will continue to do so.  We need a savegame feature added, and be able to load that back in, so players can leave & pick back up where they left off.  Hopefully this will help with later-game testing, as well.

I think using LocalStorage would be the simplest option.  I'm not sure how easily I can store all the existing blocks, whether this would be a deep copy type operation, or if simply copying the blocklist global into LocalStorage would do the trick.  Beyond the blocklist, there really isn't many variables to the game.

**Comment by mindstorm8191** [comment]
So, I've hit a few snags with generating a save & load methods, even using localStorage.  Firstly, regenerating the foods list won't work right, as doing so naturally will create 2 instances of the food items, and the ones in the actual blocks will never get eaten.  I've already set up a solution for that, in the form of a pushfoods() function in every class. Instead, we are adding the foods to the blocks, and on load, regenerating the food list.

The other problem is a bit more involved, and I haven't started working on it.  While localStorage and JSON are good at storing strings, and even objects, they unfortunately aren't built to work for classes. Every block is a class instance with a whole load of critical functions. Trying to not use classes at this point just wouldn't work.

What we'll have to do is to create a loading system that will determine what class type to load, generate that class, and then fill out all the data for that class from the rest of the stored data.  We'll have to find a way to save all this data, too. This might end up making a lot more localStorage variables, but if it works, that's no big deal.

**Comment by mindstorm8191** [comment]
I think we have this working at this point.  I was able to still use the localStorage with JSON to store the blocklist data. There is a way to re-declare an object as a class instance, so I was able to do that with the blocks and all the connected items, too (though the use of a new subclass-specific function). After a lot of other work, everything seems to be working.

I haven't fully tested this out, though.  This needs to be thoroughly bug-tested before we can consider this task done.
